modules/sizeMappingV2.js

```diff
@@ -429,57 +438,56 @@ export function getRelevantMediaTypesForBidder(sizeConfig, activeViewport) {
 
 // sets sizeMappingInternalStore for a given auctionId with relevant adUnit information returned from the call to 'getFilteredMediaTypes' function
 // returns adUnit details object.
-export function getAdUnitDetail(auctionId, adUnit) {
+export function getAdUnitDetail(auctionId, adUnit, labels) {
   // fetch all adUnits for an auction from the sizeMappingInternalStore
   const adUnitsForAuction = sizeMappingInternalStore.getAuctionDetail(auctionId).adUnits;
 
   // check if the adUnit exists already in the sizeMappingInterStore (check for equivalence of 'code' && 'mediaTypes' properties)
   const adUnitDetail = adUnitsForAuction.filter(adUnitDetail => adUnitDetail.adUnitCode === adUnit.code && utils.deepEqual(adUnitDetail.mediaTypes, adUnit.mediaTypes));
 
   if (adUnitDetail.length > 0) {
+    adUnitDetail[0].cacheHits++;
     return adUnitDetail[0];
   } else {
-    const { mediaTypes, sizeBucketToSizeMap, activeViewport, transformedMediaTypes } = internal.getFilteredMediaTypes(adUnit.mediaTypes);
+    const identicalAdUnit = adUnitsForAuction.filter(adUnitDetail => adUnitDetail.adUnitCode === adUnit.code);
+    const adUnitInstance = identicalAdUnit.length > 0 ? typeof identicalAdUnit[0].instance === 'number' ? identicalAdUnit[identicalAdUnit.length - 1].instance + 1 : 1 : 1;
+    const isLabelActivated = internal.isLabelActivated(adUnit, labels, adUnit.code, adUnitInstance);
+    const { mediaTypes = adUnit.mediaTypes, sizeBucketToSizeMap, activeViewport, transformedMediaTypes } = isLabelActivated && internal.getFilteredMediaTypes(adUnit.mediaTypes);
```

Comment:
If `isLabelActivated ` is false, then all of the destructured properties will be `undefined`. Is that correct? Could that cause any errors elsewhere?

Reply:
Yes, those properties can be left as undefined if `isLabelActivated` is false. Only exception being `medaiTypes` which I've already initialized.

This step is done for optimization. Basically the idea is, if an entire ad unit is deactivated by a failing label check, there is no need call function `getFilteredMediaTypes` and compute those values which anyway won't be used.